src/test/ledger/env.ts

```diff
@@ -85,10 +92,11 @@ export async function setupEnv(
 
 	const deployAndStore = async (deployments: typeof contractDeployments) => {
 		for (const [abi, args] of deployments) {
-			const contract = await deployContract(op, abi, args, {
+			const contract = deployContract(op, abi, args, {
 				nonce: nonce++,
 			});
-			contracts.push(contract);
+			await mineBlocks(8);
```

Comment:
You inserted a couple of `await mineBlocks(x)` calls. Shouldn't you first determine from the passed ganache options if that's even necessary and only call `mineBlocks` then?

The final `sealEpoch` call probably always makes sense though.

Reply:
In particular here, do you need to mine a block after each contract deployments because we risk that not all contract deployments fit into a single block?

Reply:
I just did not deem it necessary to switch the deployment behavior depending on whether or not the ganache is set to automine or autocommit. This way it works independent from the ganache configuration.

I would also think it is cleaner to only mine the blocks if a blocktime is specified.

And you are right regarding the extra mine calls for each contract. I did not want to risk that contract deployments exceed the gaslimit of a single block.

Reply:
Fair enough, let's keep the logic simple 👍🏻